File: Phase4/T.m
%% function for rotating coordinate axes by a given angle alpha
    function [r2]=T(r1,alpha) %r1=[x1;y1;z1]and r2=[x2;y2;z2] describe the same physical pt but in 2 coordinate systems with
        % theta degress rotation CCW
        r2=[cos(-alpha) -sin(-alpha) 0; sin(-alpha) cos(-alpha) 0;0 0 1]*r1;
        % note that we are rotating the coordinate axes, not the location vector of the particle
    end

File: Phase4/Variable_Speeds_Python/T.py
import numpy as np

def T(r1, alpha):
    rotation_matrix = np.array([
        [np.cos(-alpha), -np.sin(-alpha), 0],
        [np.sin(-alpha),  np.cos(-alpha), 0],
        [0,               0,              1]
    ])
    r2 = np.matmul(rotation_matrix, r1)
    return r2
